proposals/20220916-rules-and-plugin-distribution.md

```diff
@@ -0,0 +1,106 @@
+# Falco Rules and Plugin distribution
+This proposal aims to implement a client and related infrastructure code that allows users quickly consume and install rules and plugins distributed by the Falcosecurity organization. 
+
+This is a first step in the long-term plan to revamp `falcoctl` to make it a first citizen of the Falco ecosystem (see https://github.com/falcosecurity/falcoctl/issues/136).
+
+## Background
+
+Currently, installing **plugins** is a rather involved operation, as it requires manually finding and downloading the compiled plugin for the right architecture and then manually copying it somewhere where it can be accessed by the target Falco installation. Note that Falco could be installed in several ways (according to https://falco.org/docs/getting-started/installation/), mainly:
+* Locally on various Linux distributions
+* On Kubernetes (with Helm or by other means)
+
+Especially in the Kubernetes case, this is rather painful and involved operation since the helm chart doesn't help the user place the plugin files on the nodes.
+
+For **rules files**, the problem is similar, since it's a matter of copying a specific file into a location accessible by Falco before the main executable is started.
+
+Both rules and plugins are potentially interconnected since some ruleset require specific plugins (or a class of plugins) to exist.
+
+## Goals (in scope)
+
+We want to help DevOps/DevSecOps/Security engineers who use Falco performing the following operations:
+
+* **Search** interesting plugins and rulesets for their use case, such as monitoring cloudwatch, kubernetes logs for their managed system (EKS, GCP, ...)
+* **Specify a list of plugins and rules** (ie., an index) that need to be installed from _repositories_ (which could be managed by the Falco Organization, provided by third parties or custom, both public or private) in such a way that they could be installed locally or added to Helm configuration for easy automated deployment of a Falco installation that contain the requested plugins and rules. Additionally, this functionality could enable _subscribing_ to rule repositories that may change over time and could be updated and synced without needing to redeploy Falco.
```

Comment:
`(ie., an index)` is a bit misleading I think. This requirement is to allow the user to specify what it wants to install (which can of course use the index as correctly described below)